sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -146,14 +146,11 @@ private static bool IsOn(this ExpressionSyntax expression, SyntaxKind onKind)
             }
         }
 
-        public static bool IsInNameofCall(this ExpressionSyntax expression, SemanticModel semanticModel) =>
-            expression.Parent != null &&
-            expression.Parent.IsKind(SyntaxKind.Argument) &&
-            expression.Parent.Parent != null &&
-            expression.Parent.Parent.IsKind(SyntaxKind.ArgumentList) &&
-            expression.Parent.Parent.Parent != null &&
-            expression.Parent.Parent.Parent.IsKind(SyntaxKind.InvocationExpression) &&
-            ((InvocationExpressionSyntax)expression.Parent.Parent.Parent).IsNameof(semanticModel);
+        public static bool IsInNameofCall(this ExpressionSyntax expression, SemanticModel semanticModel)
```

Comment:
I like the code reduction, but I'm not very happy with the old `IsInNameofCall` name.  I suggest to rename it to `IsNameofArgument`. 
WDYT?

Reply:
IsInNameOfArgument? since it can be just a part of the nameof() argument